### PR TITLE
[Enhance] Refactor the import of `mmdet`

### PR DESCRIPTION
**Pull request by Xiangxu-0103**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.
Change the requirement of mmdet from first-party to third-party and modify the import accordingly.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the back-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -7,8 +7,8 @@ SPLIT_BEFORE_EXPRESSION_AFTER_OPENING_PAREN = true
 line_length = 79
 multi_line_output = 0
 extra_standard_library = setuptools
-known_first_party = mmdet,mmseg,mmdet3d
-known_third_party = cv2,imageio,indoor3d_util,load_scannet_data,lyft_dataset_sdk,m2r,matplotlib,mmcv,nuimages,numba,numpy,nuscenes,pandas,plyfile,pycocotools,pyquaternion,pytest,pytorch_sphinx_theme,recommonmark,requests,scannet_utils,scipy,seaborn,shapely,skimage,sphinx,tensorflow,terminaltables,torch,trimesh,ts,waymo_open_dataset
+known_first_party = mmdet,mmdet3d
```
Comment (ZCMax) [inline setup.cfg:10]: model should be moved to `know_third_party`

Comment (Xiangxu-0103) [reply setup.cfg:10]: Should we change the import of `mmdet` accordingly?

File: setup.cfg [hunk 2]
```diff
@@ -7,8 +7,8 @@ SPLIT_BEFORE_EXPRESSION_AFTER_OPENING_PAREN = true
 line_length = 79
 multi_line_output = 0
 extra_standard_library = setuptools
-known_first_party = mmdet,mmseg,mmdet3d
-known_third_party = cv2,imageio,indoor3d_util,load_scannet_data,lyft_dataset_sdk,m2r,matplotlib,mmcv,nuimages,numba,numpy,nuscenes,pandas,plyfile,pycocotools,pyquaternion,pytest,pytorch_sphinx_theme,recommonmark,requests,scannet_utils,scipy,seaborn,shapely,skimage,sphinx,tensorflow,terminaltables,torch,trimesh,ts,waymo_open_dataset
+known_first_party = mmdet3d
+known_third_party = cv2,imageio,indoor3d_util,load_scannet_data,lyft_dataset_sdk,m2r,matplotlib,mmcv,mmdet,mmengine,nuimages,numba,numpy,nuscenes,pandas,plyfile,pycocotools,pyquaternion,pytest,pytorch_sphinx_theme,recommonmark,requests,scannet_utils,scipy,seaborn,shapely,skimage,sphinx,tensorflow,terminaltables,torch,trimesh,ts,waymo_open_dataset
```
Comment (ZCMax) [inline setup.cfg:11]: `mmengine` should be put forward `mmdet`, from my perspective

File: setup.cfg (hunk 1 above)
Comment (ZCMax) [reply setup.cfg]: yes

File: setup.cfg (hunk 2 above)
Comment (Xiangxu-0103) [reply setup.cfg:11]: No need to put them according to the order of letters?

**Review by ZCMax — APPROVED** [review]

**Comment by ZCMax** [comment]
@ZwwWayne  Ready to be merged, please have a look at it

**Comment by Xiangxu-0103** [comment]
I have checked it with `pre-commit`, and everything is ok.

**Review by Tai-Wang — APPROVED** [review]

**Pull request merged**